kernel/comps/virtio/src/device/socket/mod.rs

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: MPL-2.0
+
+// ! #![feature(linked_list_cursors)]
+use alloc::{collections::BTreeMap, string::String, sync::Arc, vec::Vec};
+
+use aster_frame::sync::SpinLock;
+use spin::Once;
+
+use self::device::SocketDevice;
+pub mod buffer;
+pub mod config;
+pub mod connect;
+pub mod device;
+pub mod error;
+pub mod header;
+
+pub static DEVICE_NAME: &str = "Virtio-Vsock";
+pub trait VsockDeviceIrqHandler = Fn() + Send + Sync + 'static;
+
+pub fn register_device(name: String, device: Arc<SpinLock<SocketDevice>>) {
+    VSOCK_DEVICE_TABLE
+        .get()
+        .unwrap()
+        .lock()
+        .insert(name, (Arc::new(SpinLock::new(Vec::new())), device));
+}
+
+pub fn get_device(str: &str) -> Option<Arc<SpinLock<SocketDevice>>> {
+    let lock = VSOCK_DEVICE_TABLE.get().unwrap().lock();
```

Comment:
Since VSOCK_DEVICE_TABLE will be visited in irq context. All locks should be locked with irq disabled.